### PR TITLE
i#6662 public traces: add instr_t operation_size API

### PR DESCRIPTION
**Pull request by edeiana**
Adds instr_get_operation_size() and instr_set_operation_size() public
APIs for DR_ISA_REGDEPS instructions, which are currently the only
instructions for which operation_size is defined.

We set operation_size of DR_ISA_REGDEPS instructions with no
operands to OPSZ_0 in the convertion to REGDEPS process,
so decoded and converted instructions look the same.
Previously a converted instruction might have had an operation_size
different than OPSZ_0 even with no register operands, that would
then be corrected during encoding and subsequent decoding.
This is mostly for convenience when testing.

We update a test to also invoke the instr_length() API on
DR_ISA_REGDEPS instructions.

Issue #6662

### CODE REVIEW

File: api/docs/release.dox [hunk 1]
```diff
@@ -161,6 +161,8 @@ Further non-compatibility-affecting changes include:
    source and destination operands were reversed).
  - Fixed the VEX-encoded forms of vaesdec/vaesdeclast/vaesenc/vaesenclast/vpclmulqdq
    to obey VEX.L, which they previously ignored.
+ - Added instr_get_operation_size() and instr_set_operation_size() APIs for
```
Comment (derekbruening) [inline api/docs/release.dox:164]: This is in the wrong place: we put out 11.1 so this list can't be changed.  This has to be up on the post-11.1 list.

File: core/ir/instr_api.h [hunk 2]
```diff
@@ -1995,6 +1995,22 @@ DR_API
 instr_t *
 instr_convert_short_meta_jmp_to_long(void *drcontext, instrlist_t *ilist, instr_t *instr);
 
+DR_API
+/**
+ * Returns the operation size of \p instr if it's a DR_ISA_REGDEPS instruction, OPSZ_NA
```
Comment (derekbruening) [inline core/ir/instr_api.h:2000]: Use # for both DR_* and OPSZ* for links.

File: core/ir/instr_api.h [hunk 3]
```diff
@@ -1995,6 +1995,22 @@ DR_API
...
+ * otherwise.
+ */
+opnd_size_t
+instr_get_operation_size(instr_t *instr);
+
+DR_API
+/**
+ * Sets the operation size of \p instr to \p operation_size only if instr is a
+ * DR_ISA_REGDEPS instruction.
```
Comment (derekbruening) [inline core/ir/instr_api.h:2009]: # for link.

**Review by derekbruening — APPROVED** [review]

File: core/ir/instr_api.h (hunk 2 above)
Comment (edeiana) [reply core/ir/instr_api.h]: Done.

File: core/ir/instr_api.h (hunk 3 above)
Comment (edeiana) [reply core/ir/instr_api.h]: Done.

**Pull request merged**